nodes.py

```diff
@@ -181,6 +181,38 @@ def __init__(self, line_no, value):
         super().__init__(line_no, value, Types.Bool)
 
 
+class InstructionNode(Node):
```

Comment:
What's the point of this class?  I think it's redudant to Expression class. Instruction is better name though, 
So unless there are some  reasons to not do so, I sugggest to rename "Expression"  to "Instruction" (without "Node" at the end so it will match convention).